tests/testthat/test-target_market_share.R

```diff
@@ -1175,3 +1177,135 @@ test_that("`target_market_share` outputs as expected for companies with 0
   expect_equal(out$projected$technology_share, c(0, 1))
   expect_equal(out$target_sds$technology_share, c(0, 0))
 })
+
+test_that("outputs columns `percent_change_by_scope` and `scope`", {
+  out <- target_market_share(
+    fake_matched(),
+    fake_ald(),
+    fake_scenario(),
+    region_isos_stable
+  )
+
+  expected_added_columns <- c( "scope", "percentage_of_initial_production_by_scope")
+
+  expect_equal(setdiff(expected_added_columns, names(out)),character(0))
+})
+
+test_that("with known input outputs `percent_of_initial_production_by_scope` as
+          expected, at portfolio level", {
+            portfolio <- fake_matched(
+              name_ald = c("comp1", "comp2")
+            )
+
+            ald <- fake_ald(
+              technology = c("electric", "ice", "electric", "ice", "electric", "ice", "electric", "ice"),
+              year = c(2020, 2020, 2021, 2021, 2020, 2020, 2021, 2021),
+              name_company = paste0("comp", c(rep(1, 4), rep(2, 4))),
+              production = c(10, 30, 20, 20, 90, 95, 100, 100)
+            )
+
+            scenario <- fake_scenario(
+              technology = c("electric", "ice", "electric", "ice"),
+              year = c(2020, 2020, 2021, 2021),
+              tmsr = c(1, 1, 1.85, 0.6),
+              smsp = c(0, 0, 0.34, -0.2)
+            )
+
+            out <- target_market_share(
+              portfolio,
+              ald,
+              scenario,
+              region_isos_stable,
+              by_company = FALSE,
+              weight_production = FALSE
+            )
+
+            out_percent <- out %>%
+              filter(.$year == 2021) %>%
+              split(.$metric)
+
+            expect_equal(
```

Comment:
`testthat::expect_equal` has a `tolerance` argument that could make things easier here (as opposed to rounding)